xwiki-platform-core/xwiki-platform-mentions/xwiki-platform-mentions-api/src/main/java/org/xwiki/mentions/notifications/MentionNotificationParameters.java

```diff
@@ -19,105 +19,153 @@
-        DocumentReference mentionedIdentity, MentionLocation location, String anchorId, XDOM xdom)
+    public MentionNotificationParameters(String authorReference,
+        EntityReference entityReference, MentionLocation location, String version)
     {
         this.authorReference = authorReference;
-        this.documentReference = documentReference;
-        this.mentionedIdentity = mentionedIdentity;
+        this.entityReference = entityReference;
         this.location = location;
-        this.anchorId = anchorId;
-        this.xdom = xdom;
+        this.version = version;
     }
 
     /**
-     *
      * @return the reference of the author of the mention
      */
-    public DocumentReference getAuthorReference()
+    public String getAuthorReference()
     {
         return this.authorReference;
     }
 
     /**
-     *
      * @return the document in which the mention has been done
      */
-    public DocumentReference getDocumentReference()
+    public EntityReference getEntityReference()
     {
-        return this.documentReference;
+        return this.entityReference;
     }
 
     /**
-     *
-     * @return the identity of the mentioned user
+     * @return the location of the mention
      */
-    public DocumentReference getMentionedIdentity()
+    public MentionLocation getLocation()
     {
-        return this.mentionedIdentity;
+        return this.location;
     }
 
     /**
+     * Add a to the map of mentions.
      *
-     * @return the location of the mention
+     * @param type the type of the mentioned user
+     * @param mentionedActorReference the mention notification parameters
+     * @return the current object
      */
-    public MentionLocation getLocation()
+    public MentionNotificationParameters addMention(String type,
+        MentionNotificationParameter mentionedActorReference)
     {
-        return this.location;
+        addToMap(type, mentionedActorReference, this.mentions);
+        return this;
     }
 
     /**
+     * Add a mention to the map of new mentions.
      *
-     * @return the anchor link to use
+     * @param type the type of the mentioned user
+     * @param mentionedActorReference the mention notification parameter
+     * @return the current object
      */
-    public String getAnchorId()
+    public MentionNotificationParameters addNewMention(String type,
+        MentionNotificationParameter mentionedActorReference)
     {
-        return this.anchorId;
+        addToMap(type, mentionedActorReference, this.newMentions);
+        return this;
     }
 
     /**
+     * @return the map of new mentions
+     */
+    public Map<String, Set<MentionNotificationParameter>> getNewMentions()
+    {
+        return this.newMentions;
+    }
+
+    /**
+     * @return the map of all the mentions
+     */
+    public Map<String, Set<MentionNotificationParameter>> getMentions()
```

Comment:
Same comment for the key of the map. I guess all mentions include new mentions too? Wouldn't that be better to have newMentions and oldMentions? I'm not sure about the usecases here.

Reply:
Updated as well.